core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/provision/ResourceManifestGeneratorImpl.java

```diff
@@ -45,42 +47,31 @@ public void registerProviderGenerator(ResourceDefinitionGenerator generator) {
     }
 
     @Override
-    public void registerPermissionFunctions(Map<String, AtomicConstraintFunction<String, Permission, Boolean>> functions) {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public void registerProhibitionFunctions(Map<String, AtomicConstraintFunction<String, Prohibition, Boolean>> functions) {
-        throw new UnsupportedOperationException();
+    public ResourceManifest generateResourceManifest(TransferProcess process) {
+        var definitions = definitions(process);
+        return ResourceManifest.Builder.newInstance().definitions(definitions).build();
     }
 
-    @Override
-    public void registerObligationFunctions(Map<String, AtomicConstraintFunction<String, Duty, Boolean>> functions) {
-        throw new UnsupportedOperationException();
+    private List<ResourceDefinition> definitions(TransferProcess process) {
```

Comment:
Minor nit: can we name this `createDefinitions` or something similar, so that it becomes clear what it does?

Reply:
ok